sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/E2ETelemetryItemValidation/TracesTests.cs

```diff
@@ -280,14 +278,14 @@ public void TestActivityEvents()
                 .Build();
 
             // ACT
-            string spanId = null, traceId = null;
+            string spanId, traceId;
 
-            using (var activity = activitySource.StartActivity(name: "ActivityWithException"))
+            using (var activity = activitySource.StartActivity(name: "ActivityWithException") ?? throw new Exception("Failed to create Activity"))
```

Comment:
I see this in many places, why do we need to throw when activity is null. We don't have any tests to validate the scenario. In general activity could be null if it does not have an Listener. These tests might give a different understanding about activities for customer.

Reply:
`StartActivity()` can return null if nothing is subscribed to the ActivitySource.
This was the easiest way to deal with the null and satisfy the compiler.
Here the exception is valid... if an activity isn't created, the test will fail.

The alternative is to decorate every interaction with the activity with null-conditional `?.` and null-forgiving `!` operators. 


These tests could possibly be rewritten to refactor this as boilerplate code.

Reply:
Not a strong preference but I would vote for `?`. IIRC visual studio will give suggestion to add `?` if someone missed that. Does it give suggestion to add throw as well?

Reply:
Visual Studio did recommend these at first, but it was A LOT of `?.` and `!` decorations.
There was also some inconsistency which made the rest of the method look odd. 
Any direct member (property or method) required the null-conditional `?.`. But some extension methods do their own null check so the null-conditional wasn't needed.
And anywhere we pass the activity to a method required either the null-forgiving `!` operator or some additional null check.

I felt that this was the simplest option to address all cases.

Reply:
Discussed offline. 
Will remove the "throw" from this PR and will review options in a follow up.

Reply:
After updating xunit, I was able to fix this with `Assert.NotNull`.
The newest version adds NullableAttributes to their implementation.